lib/medusa/index.ts

```diff
@@ -0,0 +1,337 @@
+import { isMedusaError } from 'lib/type-guards';
+
+import { mapOptionIds } from 'lib/utils';
+import {
+  Cart,
+  CartItem,
+  MedusaCart,
+  MedusaLineItem,
+  MedusaProduct,
+  MedusaProductCollection,
+  MedusaProductOption,
+  MedusaProductVariant,
+  Product,
+  ProductCollection,
+  ProductOption,
+  ProductVariant,
+  SelectedOption
+} from './types';
+
+// const endpoint = `${process.env.MEDUSA_BACKEND_API!}`;
+const endpoint = `http://localhost:9000/store`;
+
+export default async function medusaRequest(
+  method: string,
+  path = '',
+  payload?: Record<string, unknown> | undefined
+) {
+  const options: RequestInit = {
+    method,
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  };
+
+  if (payload) {
+    options.body = JSON.stringify(payload);
+  }
+
+  try {
+    const result = await fetch(`${endpoint}/${path}`, options);
+
+    const body = await result.json();
+
+    if (body.errors) {
+      throw body.errors[0];
+    }
+
+    return {
+      status: result.status,
+      body
+    };
+  } catch (e) {
+    if (isMedusaError(e)) {
+      throw {
+        status: e.status || 500,
+        message: e.message
+      };
+    }
+
+    throw {
+      error: e
+    };
+  }
+}
+
+const reshapeCart = (cart: MedusaCart): Cart => {
+  const lines = cart?.items?.map((item) => reshapeLineItem(item)) || [];
+  const totalQuantity = lines.length;
+  const checkoutUrl = '/';
+  const currencyCode = 'EUR';
+  const cost = {
+    subtotalAmount: {
+      amount:
+        (cart?.tax_total && cart?.total && (cart?.total - cart?.tax_total).toString()) ||
+        cart?.total?.toString() ||
+        '0',
+      currencyCode: currencyCode
+    },
+    totalAmount: {
+      amount: (cart?.total && cart?.total.toString()) || '0',
+      currencyCode: currencyCode
+    },
+    totalTaxAmount: {
+      amount: (cart?.tax_total && cart?.tax_total.toString()) || '0',
+      currencyCode: currencyCode
+    }
+  };
+
+  return {
+    ...cart,
+    totalQuantity,
+    checkoutUrl,
+    lines,
+    cost
+  };
+};
+
+const reshapeLineItem = (lineItem: MedusaLineItem): CartItem => {
+  const product = {
+    title: lineItem.title,
+    priceRange: {
+      maxVariantPrice: {
+        amount: lineItem.variant?.prices?.[0]?.amount.toString() ?? '0',
+        currencyCode: lineItem.variant?.prices?.[0]?.currency_code ?? 'EUR'
+      }
+    },
+    updatedAt: lineItem.updated_at,
+    tags: [],
+    descriptionHtml: lineItem.description ?? '',
+    featuredImage: {
+      url: lineItem.thumbnail ?? '',
+      altText: lineItem.title ?? ''
+    },
+    availableForSale: true,
+    variants: [lineItem.variant && reshapeProductVariant(lineItem.variant)],
+    handle: lineItem.variant?.product?.handle ?? ''
+  };
+
+  const selectedOptions =
+    lineItem.variant?.options?.map((option) => ({
+      name: option.option?.title ?? '',
+      value: option.value
+    })) || [];
+
+  const merchandise = {
```

Comment:
question: What is this property used for? Maybe we can change its name to something within the Medusa realm?

Reply:
Cool to keep it as is 👍